### PR TITLE
update for new `AlgorithmInterpretation`

### PR DESCRIPTION
**Pull request by prbzrg**
Related to
https://github.com/SciML/SciMLBase.jl/pull/777
https://github.com/SciML/StochasticDiffEq.jl/pull/583

### CODE REVIEW

File: src/sde_default_alg.jl [hunk 1]
```diff
@@ -13,16 +13,17 @@ function default_algorithm(
     end
 
     is_stiff = :stiff ∈ alg_hints
-    is_stratonovich = :Stratonovich ∈ alg_hints
+    is_stratonovich = :stratonovich ∈ alg_hints
```
Comment (ChrisRackauckas) [inline src/sde_default_alg.jl:16]: why this change?

Comment (prbzrg) [reply src/sde_default_alg.jl:16]: Because of docs at
https://github.com/SciML/DiffEqBase.jl/blob/13ac2da52c44494a57132245a3c6baec1c9cbb0e/src/solve.jl#L758

**Comment by ChrisRackauckas** [comment]
This is going to need to bump the minor on SciMLBase and StochasticDiffEq, but other than that looks good to go.

**Comment by ChrisRackauckas** [comment]
@oscardssmith did you never update the DiffEq tests with the default change?

**Comment by oscardssmith** [comment]
Oh, I guess I never did. Will fix.

**Comment by prbzrg** [comment]
Is there any reason to keep codes related to ODE?

https://github.com/SciML/DifferentialEquations.jl/blob/master/src/ode_default_alg.jl

https://github.com/SciML/DifferentialEquations.jl/blob/master/test/default_ode_alg_test.jl

**Comment by oscardssmith** [comment]
I don't think so.

**Comment by ChrisRackauckas** [comment]
Are the same tests in OrdinaryDiffEq? If so then yes let's just delete.

**Comment by ChrisRackauckas** [comment]
Rebase this and tests should pass.

**Comment by ChrisRackauckas** [comment]
@ErikQQY looks like some depwarns crept back in? https://github.com/SciML/DifferentialEquations.jl/actions/runs/10668385943/job/29567982165?pr=1048#step:6:584

We can merge this though.
